### PR TITLE
Fix panic bug in snapshot inspect

### PR DESCRIPTION
**Pull request by banks**
While working on #10089 I realised that currently the command will panic if passed a non-snapshot file due to a missing return.

```
consul snapshot inspect ./consul-data/s1/raft/snapshots/9-4600669-1618935304715/meta.json
Error reading snapshot: failed to decompress snapshot: gzip: invalid header
Error extracting snapshot data: invalid argument
Failed to close temp snapshot: invalid argument
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x356daa1]

goroutine 1 [running]:
os.(*File).Name(...)
	/usr/local/Cellar/go/1.15.7/libexec/src/os/file.go:54
github.com/hashicorp/consul/command/snapshot/inspect.(*cmd).Run.func1(0x0, 0xc0000b1920)
	/Users/banks/src/hashicorp/consul/command/snapshot/inspect/snapshot_inspect.go:122 +0xc1
github.com/hashicorp/consul/command/snapshot/inspect.(*cmd).Run(0xc0000b1920, 0xc00004c0b0, 0x1, 0x1, 0x1)
	/Users/banks/src/hashicorp/consul/command/snapshot/inspect/snapshot_inspect.go:130 +0x420
github.com/mitchellh/cli.(*CLI).Run(0xc00012a000, 0xc00012a000, 0x80, 0xc0002ba340)
	/Users/banks/src/go/pkg/mod/github.com/mitchellh/cli@v1.1.0/cli.go:260 +0x1cf
main.realMain(0xc000054118)
	/Users/banks/src/hashicorp/consul/main.go:49 +0x43d
main.main()
	/Users/banks/src/hashicorp/consul/main.go:21 +0x25
```

This adds a (previously failing due to panic) test to cover it too!

### CODE REVIEW

**Comment by mkeeler** [comment]
FYI I added backport labels for 1.7, 1.8 and 1.9. If they don't apply cleanly then we could probably ignore it as a user could use the CLI of a newer version with the an older snapshot.

File: .changelog/10091.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+cli: snapshot inspect command would panic on invalid input. [[GH-10091](https://github.com/hashicorp/consul/pull/10091)]
```
Comment (rboyer) [inline .changelog/10091.txt:2]: Doesn't the changelog-build tool auto-add the `[GH-xxxx]` part already?

Comment (banks) [reply .changelog/10091.txt:2]: Oh, I didn't know that - some examples I saw still have that explicitly but good to know. I'll remove them.

Comment (banks) [inline .changelog/10091.txt:2]:
```suggestion
cli: snapshot inspect command would panic on invalid input.
```

**Pull request merged**